### ISSUE TITLE
Exception on CONTAINS statement in SUBMODULE PROCEDURE

### ISSUE THREAD

**Issue by rouson** [Fortran-FOSS-Programmers/ford#326]
Running `ford` on the two-line project file below throws an exception due to an internal subprogram inside a submodule subroutine:
```
% ford --version
FORD, version 6.0.0

% cat internal-procedure-bug.md     

project: internal-procedure-bug
src_dir: src

% cat src/internal-procedure-bug.f90
```
```fortran
module foo_m
  implicit none
  interface
    module subroutine foo()
      implicit none
    end subroutine
  end interface
end module

submodule(foo_m) foo_s
  implicit none
contains
  module procedure foo
  contains
    subroutine bar()
    end subroutine
  end procedure
end submodule
```
```
 % ford internal-procedure-bug.md 
Reading file src/internal-procedure-bug.f90
Traceback (most recent call last):
  File "/usr/local/bin/ford", line 8, in <module>
    sys.exit(run())
  File "/usr/local/lib/python3.9/site-packages/ford/__init__.py", line 391, in run
    main(proj_data,proj_docs,md)
  File "/usr/local/lib/python3.9/site-packages/ford/__init__.py", line 342, in main
    project = ford.fortran_project.Project(proj_data)
  File "/usr/local/lib/python3.9/site-packages/ford/fortran_project.py", line 90, in __init__
    ford.sourceform.FortranSourceFile(os.path.join(curdir,item),settings, preprocessor, ext in self.fixed_extensions,incl_src=html_incl_src))
  File "/usr/local/lib/python3.9/site-packages/ford/sourceform.py", line 1052, in __init__
    FortranContainer.__init__(self,source,"")
  File "/usr/local/lib/python3.9/site-packages/ford/sourceform.py", line 653, in __init__
    self.submodules.append(FortranSubmodule(source,
  File "/usr/local/lib/python3.9/site-packages/ford/sourceform.py", line 628, in __init__
    self.modprocedures.append(FortranSubmoduleProcedure(source,
  File "/usr/local/lib/python3.9/site-packages/ford/sourceform.py", line 560, in __init__
    raise Exception("CONTAINS statement in {}".format(type(self).__name__[7:].upper()))
Exception: CONTAINS statement in SUBMODULEPROCEDURE
```

**Comment by rouson** [comment]
FWIW, there's a relatively painless [workaround](https://github.com/sourceryinstitute/sourcery/blob/b5331415ec7fb9f78ee5acc7e0396d2ac15b68ff/src/data-partition-implementation.F90#L31).

**Comment by ZedThree** [comment]
Thanks for the reproducer @rouson, I've added it to the unit tests I'm writing.

Fixed in #321
